core/arch/arm/tee/entry_std.c

```diff
@@ -379,7 +489,17 @@ void __weak tee_entry_std(struct thread_smc_args *smc_args)
 	}
 	parg = (uint64_t)smc_args->a1 << 32 | smc_args->a2;
 
-	mobj = get_cmd_buffer(parg, &num_params);
+	/* Check if this region is in static shared space */
+	if (core_pbuf_is(CORE_MEM_NSEC_SHM, parg,
+			  sizeof(struct optee_msg_arg))) {
+		mobj = get_cmd_buffer(parg, &num_params);
+	} else {
+		if (parg & SMALL_PAGE_MASK) {
+			smc_args->a0 = OPTEE_SMC_RETURN_EBADADDR;
+			return;
+		}
+		mobj = map_cmd_buffer(parg, &num_params);
+	}
```

Comment:
I prefer to leave it in current state. `get_cmd_buffer()` is responsible for buffers in `MEM_AREA_NSEC_SHM` which is already mapped in OP-TEE address space. `map_cmd_buffer()` maps buffers from `MEM_AREA_RAM_NSEC`. One task for one function.

As alternative I can create a third function that will choose to call  `get_cmd_buffer()` or `map_cmd_buffer()`.